packages/lodestar/src/network/gossip/gossipsub.ts

```diff
@@ -151,31 +149,32 @@ export class Eth2Gossipsub extends Gossipsub {
     return msgId;
   }
 
+  // no need to overwrite _processMessages thanks to this https://github.com/libp2p/js-libp2p-interfaces/blob/libp2p-interfaces%404.0.4/packages/interfaces/src/pubsub/index.js#L367
   // Temporaly reverts https://github.com/libp2p/js-libp2p-interfaces/pull/103 while a proper fixed is done upstream
   // await-ing _processRpc causes messages to be processed 10-20 seconds latter than when received. This kills the node
-  async _processMessages(
-    idB58Str: string,
-    stream: AsyncIterable<Uint8Array | BufferList>,
-    peerStreams: PeerStreams
-  ): Promise<void> {
-    try {
-      await pipe(stream, async (source) => {
-        for await (const data of source) {
-          const rpcBytes = data instanceof Uint8Array ? data : data.slice();
-          const rpcMsg = this._decodeRpc(rpcBytes);
-
-          this._processRpc(idB58Str, peerStreams, rpcMsg).catch((e) => {
-            this.log("_processRpc error", (e as Error).stack);
-          });
-        }
-      });
-    } catch (err) {
-      this._onPeerDisconnected(peerStreams.id, err as Error);
-    }
-  }
+  // async _processMessages(
```

Comment:
need some confirmation before removing this function completely

Reply:
This is the override that suppress the queue in gossipsub right?

Reply:
No this override was originally used to stop this behavior:
```ts
for await (const bytes of source) {
  await this.processRpc(decodeBytes(bytes));
}
```

But the recent upstream now essentially does whats commented out below:
```ts
for await (const bytes of source) {
  this.processRpc(decodeBytes(bytes)).catch(...);
}
```